### DISCUSSION TITLE
Open-to-flow boundary condition implementation.

### DISCUSSION THREAD

**Discussion by anupkgeos** [idaholab/moose#25884] [Q&A Modules: Porous Flow] [answered]
Hi
I want to solve CO2 infection problem in a brine-saturated reservoir: 
![cappa_2011a-Page-1 drawio(1)](https://github.com/idaholab/moose/assets/145852108/457dee53-348a-4550-9a45-5fd189c50ef6)

1. How can implement open-to-flow boundaries as:
![image](https://github.com/idaholab/moose/assets/145852108/87b9280e-656c-4f08-87d0-b42adeaa54ae)
Here primary variables are `pgas, zi, and xnacl.`  Initially, the reservoir is fully saturated with brine(Xnacl=0.1). 

```
[BCs]
  # Left pgas = No flow
  [top_pgas]
    type = DirichletBC
    variable = pgas
    value = 5e6  # 5 MPa
    boundary = 'top'
  []
  [top_zi]
    type = DirichletBC   # or PorousFlowOutflowBC ??
    variable = zi
    value = 0  
    boundary = 'top'
  []
  [top_xnacl]
    type = DirichletBC  # or PorousFlowOutflowBC ??
    variable = xnacl
    value = 0.1 
    boundary = 'top'
  []
[]

```
For outflow, which one would be more appropriate here: `DirichletBC  # or PorousFlowOutflowBC ??`

Thank you!

**Comment by GiudGiud** [comment]
Hello

DirichletBC are appropriate for pressure at the top I think.
For flow I would not expect to set the concentration on the boundary. Especially free-flow. 
@cpgr may want to pitch in, though did you check if there's an example of this in the module?

Guillaume

**Reply by cpgr** [reply, marked as answer]
I would try Dirichlet for pressure and temperature at the top and bottom (and I guess a function Dirichlet on the right hand side), and the PorousFlowOutflowBC for the other variables. I'm not sure what they mean by constant saturation though - if gas phase CO2 can reach a boundary and flow out then the saturation will change at that boundary?

**Reply by anupkgeos** [reply]
In the paper pgas and sgas are used as the primary variable and a constant saturation is applied. 
I tried the BCs as you mentioned...simulation runs okay for a few steps but it seems CO2 saturation fronts don't move forward and pressure increases too much and at the end getting pressure must be less than 60 MPa and non-linear solution doesn't converge. Here is my input and mesh file...would you please have a look?

https://github.com/anupkgeos/moose_co2_test/blob/main/5cappa2011a_2comp_simple.i
https://github.com/anupkgeos/moose_co2_test/blob/main/cappa2011_simple.msh

**Reply by cpgr** [reply]
Sorry about the late reply! I think your injection rate is too high - it is integrated over the boundary. Did you mean for this to be a cylindrical model? At the moment, it is only 2D.
